packages/playback-core/src/index.ts

```diff
@@ -566,12 +567,19 @@ export const loadMedia = (
       },
       { once: true }
     );
-    addEventListenerWithTeardown(mediaEl, 'pause', () => {
-      if (!getEnded(mediaEl)) return;
+    const maybeDispatchEndedCallback = () => {
+      // Do the "cheaper" check first
       if (mediaEl.ended) return;
```

Comment:
non-blocking: is this line needed? it looks like `getEnded()` does some very cheap checks and `mediaEl.ended` is evaluated before `isPseudoEnded()` in that function also. Just wondering if the redundancy is needed. Either way it's fine 🙂